format

```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+find src -type f | egrep -v '(lib\/.*\/|linux|archives)' | egrep '\.[ch]$' | xargs clang-format -i
```

Comment:
We don't want to run it on some libraries, but we do on others.  This feels like it should be built into the build system so that we can control this.  Perhaps:

`make format`